prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -446,17 +443,15 @@ private static void validateExtensionName(final String extensionName) {
 
     private static JSONArray buildEnumerateResult(final List<String> extensions) throws FalconException {
         JSONArray results = new JSONArray();
-
-        for (String extension : extensions) {
-            String extensionType = AbstractExtension.isExtensionTrusted(extension) ? TRUSTED_EXTENSION
-                    : CUSTOM_EXTENSION;
-
+        ExtensionMetaStore metricStore = ExtensionStore.get().getMetricStore();
+        List<ExtensionMetadataBean> beanList = metricStore.getAllExtensions();
+        for (ExtensionMetadataBean bean : beanList) {
             JSONObject resultObject = new JSONObject();
 
             try {
-                resultObject.put(EXTENSION_NAME, extension.toLowerCase());
-                resultObject.put(EXTENSION_TYPE, extensionType);
-                resultObject.put(EXTENSION_DESC, getShortDescription(extension));
+                resultObject.put(EXTENSION_NAME, bean.getExtensionName().toLowerCase());
+                resultObject.put(EXTENSION_TYPE, bean.getExtensionType());
+                resultObject.put(EXTENSION_DESC, bean.getDescription());
```

Comment:
Think @sandeepSamudrala mentioned in the another pull request. Should add location to the resultObject.